src/AbstractSQLCompiler.ts

```diff
@@ -586,19 +587,46 @@ $$ LANGUAGE ${fnDefinition.language};`);
 					_.includes(['ForeignKey', 'ConceptType'], dataType) &&
 					references != null
 				) {
-					foreignKeys.push({ fieldName, references });
-					depends.push(references.resourceName);
-					hasDependants[references.resourceName] = true;
+					const referencedTable =
+						abstractSqlModel.tables[references.resourceName];
+					const fkDefinition = `FOREIGN KEY ("${fieldName}") REFERENCES "${referencedTable.name}" ("${references.fieldName}")`;
+
+					const schemaInfo = schemaDependencyMap[references.resourceName];
+					if (schemaInfo && schemaInfo.depends.includes(table.resourceName)) {
+						// It's a simple circular dependency so we switch it to an ALTER TABLE
+						// TODO: Remove the postgres specific "IF NOT EXISTS" check
+						alterSchemaStatements.push(`\
+DO $$
+BEGIN
+	IF NOT EXISTS (
```

Comment:
What are the cases this foreign key can already exist?

Reply:
The create schema statements are rerun on pinejs startup to create any new tables that are missing (or in the case of a new env to create everything) so in those cases the foreign keys can already exist and is why all of the create statements include the `IF NOT EXISTS` equivalent